packages/state-manager/src/sagas/identity/saveUserCsr/saveUserCsr.saga.test.ts

```diff
@@ -0,0 +1,123 @@
+import { combineReducers } from '@reduxjs/toolkit'
+import { expectSaga } from 'redux-saga-test-plan'
+import { type Socket } from '../../../types'
+import { communitiesReducer, CommunitiesState, type communitiesActions } from '../../communities/communities.slice'
+import { StoreKeys } from '../../store.keys'
+import { identityAdapter } from '../identity.adapter'
+import { identityReducer, IdentityState, type identityActions } from '../identity.slice'
+import { saveUserCsrSaga } from './saveUserCsr.saga'
+import { type Store } from '../../store.types'
+import { type FactoryGirl } from 'factory-girl'
+import { setupCrypto } from '@quiet/identity'
+import { prepareStore } from '../../../utils/tests/prepareStore'
+import { getFactory } from '../../../utils/tests/factories'
+import { SocketActionTypes } from '@quiet/types'
+import { usersReducer, UsersState } from '../../users/users.slice'
+import { certificatesAdapter } from '../../users/users.adapter'
+
+describe('saveUserCsr', () => {
+  let store: Store
+  let factory: FactoryGirl
+
+  beforeEach(async () => {
+    setupCrypto()
+    store = prepareStore().store
+    factory = await getFactory(store)
+  })
+
+  test('saves user csr if user csr is absent in csrs list', async () => {
+    const socket = { emit: jest.fn(), on: jest.fn() } as unknown as Socket
+
+    const community =
+      await factory.create<ReturnType<typeof communitiesActions.addNewCommunity>['payload']>('Community')
+
+    const identity = await factory.create<ReturnType<typeof identityActions.addNewIdentity>['payload']>('Identity', {
+      id: community.id,
+      nickname: 'john',
+    })
+
+    await expectSaga(saveUserCsrSaga, socket)
+      .withReducer(
+        combineReducers({
```

Comment:
This is definitely a deprecated way of mocking reducers. You're actually not gaining any value from using `factory.create`. 
Try
```
const reducer = combineReducers(reducers)

.withReducer(reducer)
.withState(store.getState())
```

Reply:
Unfortunately in the first test case factory side effects are unwanted (putting csr into store)